platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/builder/PlatformBuilder.java

```diff
@@ -288,6 +293,19 @@ public PlatformBuilder withStaleEventCallback(@NonNull final Consumer<PlatformEv
     public PlatformBuilder withKeysAndCerts(@NonNull final KeysAndCerts keysAndCerts) {
         throwIfAlreadyUsed();
         this.keysAndCerts = Objects.requireNonNull(keysAndCerts);
+        // Ensure that the platform has a valid signing cert that matches the signing private key.
```

Comment:
this does not check if it matches the private key

Reply:
lines 302-307 signs a test value and verifies it.  Wouldn't that ensure the keys match?

Reply:
ah, yes, I missed that